demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+                break;
+            }
+        }
+
+        if( pIndex == NULL )
+        {
+            /* Fail if no connection could be established. */
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "Could not connect to any resolved IP address from %.*s.\n",
+                        ( int ) strlen( pServer ),
+                        pServer ) );
+        }
+        else
+        {
+            returnStatus = EXIT_SUCCESS;
+            LogInfo( ( "TCP connection established with %.*s.\n",
+                       ( int ) strlen( pServer ),
+                       pServer ) );
+        }
+    }
+    else
+    {
+        LogError( ( "Could not resolve host %.*s.\n",
+                    ( int ) strlen( pServer ),
+                    pServer ) );
+        returnStatus = EXIT_FAILURE;
+    }
+
+    /* Set the socket option for send and receive timeouts. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        transportTimeout.tv_sec = 0;
+        transportTimeout.tv_usec = ( TRANSPORT_SEND_RECV_TIMEOUT_MS * 1000 );
+
+        /* Set the receive timeout. */
+        if( setsockopt( *pTcpSocket,
+                        SOL_SOCKET,
+                        SO_RCVTIMEO,
+                        ( char * ) &transportTimeout,
+                        sizeof( transportTimeout ) ) < 0 )
+        {
+            LogError( ( "Setting socket receive timeout failed." ) );
+            returnStatus = EXIT_FAILURE;
+        }
+
+        /* Set the send timeout. */
+        if( setsockopt( *pTcpSocket,
+                        SOL_SOCKET,
+                        SO_SNDTIMEO,
+                        ( char * ) &transportTimeout,
+                        sizeof( transportTimeout ) ) < 0 )
+        {
+            LogError( ( "Setting socket send timeout failed." ) );
+            returnStatus = EXIT_FAILURE;
+        }
+    }
+
+    if( pListHead != NULL )
+    {
+        freeaddrinfo( pListHead );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportSend( HTTPNetworkContext_t * pContext,
+                              const void * pBuffer,
+                              size_t bytesToSend )
+{
+    int32_t bytesSend = 0;
+
+    bytesSend = send( ( int ) pContext->tcpSocket, pBuffer, bytesToSend, 0 );
+
+    if( bytesSend < 0 )
+    {
+        /* Check if it was time out */
+        if( ( errno == EAGAIN ) || ( errno == EWOULDBLOCK ) )
+        {
+            /* Set return value to 0 to indicate that send was timed out. */
+            bytesSend = 0;
+        }
+    }
+
+    return bytesSend;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportRecv( HTTPNetworkContext_t * pContext,
+                              void * pBuffer,
+                              size_t bytesToRecv )
+{
+    int32_t bytesReceived = 0;
+
+    bytesReceived = ( int32_t ) recv( ( int ) pContext->tcpSocket, pBuffer, bytesToRecv, 0 );
```

Comment:
Both `int32_t` and `int` casts seem to be redundant

Reply:
Took out `int` cast, but `int32_t` cast required by definition of transport receive.

Reply:
The local variable `bytesReceived` is of `int32_t` already. A cast should not be _explicitly_ required on the return value